integration-tests/ahm/src/tests.rs

```diff
@@ -653,11 +665,24 @@ async fn migration_works_time() {
 		AhMigrationStageStorage::<AssetHub>::put(AhMigrationStage::DataMigrationOngoing);
 	});
 
-	let ah_block_start = ah.execute_with(frame_system::Pallet::<AssetHub>::block_number);
+	let new_ah_block_start = ah.execute_with(frame_system::Pallet::<AssetHub>::block_number);
 
 	log::info!("Running the migration second time");
 
-	migrate(ah_block_start, &mut rc, &mut ah);
+	migrate(new_ah_block_start, &mut rc, &mut ah);
+
+	let new_rc_block_end = rc.execute_with(frame_system::Pallet::<Polkadot>::block_number);
+	let new_ah_block_end = ah.execute_with(frame_system::Pallet::<AssetHub>::block_number);
+
+	rc.execute_with(|| {
+		assert_eq!(RcMigrationStartBlock::<Polkadot>::get(), Some(rc_block_start + 1));
+		assert_eq!(RcMigrationEndBlock::<Polkadot>::get(), Some(new_rc_block_end));
+	});
+
+	ah.execute_with(|| {
+		assert_eq!(AhMigrationStartBlock::<AssetHub>::get(), Some(ah_block_start + 1));
+		assert_eq!(AhMigrationEndBlock::<AssetHub>::get(), Some(new_ah_block_end));
```

Comment:
Not sure how to handle the idempotent case; but I guess there is no easy answer here.

Reply:
Ah yea, i forgot that we wanted to have this cooloff stage where we can rerun some stages. 👍